### ISSUE TITLE
api-version processing changes based on header vs query

### ISSUE THREAD

**Issue by pakrym** [Azure/autorest.modelerfour#339]
``` json
 "parameters": {
        "ApiVersionParameter": {
            "in": "query",
            "name": "api-version",
            "description": "Version of API to invoke",
            "required": true,
            "type": "string"
        },
}
```
Results in:
```yml
 - !<!Parameter> &ref_126
    schema: *ref_123
    implementation: Client
    origin: 'modelerfour:synthesized/api-version'
    required: true
    language: !<!Languages> 
      default:
        name: apiVersion
        description: Api Version
        serializedName: api-version
    protocol: !<!Protocols> 
      http: !<!HttpParameter> 
        in: query
    - !<!ConstantSchema> &ref_123
      type: constant
      value: !<!ConstantValue> 
        value: 2020-07-20-preview1
      valueType: *ref_1
      language: !<!Languages> 
        default:
          name: ApiVersion20200720Preview1
          description: Api Version (2020-07-20-preview1)
      protocol: !<!Protocols> {}
```
But 
``` json
 "parameters": {
        "ApiVersionParameter": {
            "in": "header",
            "name": "api-version",
            "description": "Version of API to invoke",
            "required": true,
            "type": "string"
        },
}
```

Results in:
```yml
 - !<!Parameter> &ref_125
    schema: *ref_123
    implementation: Client
    required: true
    extensions:
      x-ms-priority: 0
    language: !<!Languages> 
      default:
        name: ApiVersion
        description: Version of API to invoke
        serializedName: api-version

 - !<!StringSchema> &ref_123
      type: string
      apiVersions:
        - !<!ApiVersion> 
          version: 2020-07-20-preview1
      language: !<!Languages> 
        default:
          name: String
          description: ''
      protocol: !<!Protocols> {}

```

**Comment by pakrym** [comment]
cc @daviwil

**Comment by daviwil** [comment]
This is how modelerfour determines whether a parameter might actually refer to the service version instead of being an arbitrary pararameter named `api-version`.  You can see the check in the code here:

https://github.com/Azure/autorest.modelerfour/blob/cdf9aa2d19bdc17b879d7bcad0aa4a2482b98eeb/modelerfour/modeler/interpretations.ts#L154-L162

Anything that isn't a query parameter is automatically excluded from the check.  After that, `x-ms-api-version` is checked to determine whether this parameter should be treated as a service API version parameter.  If its value is `false`, it's treated as a normal parameter.

**Comment by daviwil** [comment]
Fix released in `4.15.416`, please close once you've verified that the fix resolves the issue for you.  Thanks!
